### ISSUE TITLE
cloudimpl: Segfault when creating scheduled backup to GCS in v20.2.2

### ISSUE THREAD

**Issue by a-robinson** [cockroachdb/cockroach#57968]
**Describe the problem**

When I try to set up a backup schedule to GCS on a local single-node instance, it pretty reliably crashes the instance with a sigsegv.

**To Reproduce**

1. Start up a one-node instance using `cockroach start-single-node`. There's some data in it, but not much, and I don't know whether that matters. It's running with `--insecure` and `--listen-addr localhost`, although I doubt either of those matters. My node's default zone config has `gc.ttlseconds` set to just 10 minutes instead of the default, although again I doubt that makes a difference.
2. Add GCS credentials using the `cloudstorage.gs.default.key` cluster setting as instructed in the docs
3. Confirm that you can successfully run a single backup: `backup to 'gs://my-bucket-name/test1' as of system time '-30s';`. This should work.
4. Try setting up a backup schedule to the same bucket from the same instance to make its first run just a couple minutes in the future: `create schedule daily_backup for backup into 'gs://my-bucket-name/test-daily' recurring '@daily' with schedule options first_run = '2020-12-15 20:48:00+00:00';`. In reproducing, you may want to make sure you set the timestamp to be in the future.

This fails, killing cockroach and disconnecting my sql shell. In the stderr log file from the cockroach instance, I get this stack trace, which interestingly enough suggests the bug may be in the GCS client library rather than cockroach:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xc1d442]

goroutine 1623149 [running]:
net/http.(*Client).deadline(0x0, 0x498f720, 0xc0056a1080, 0x80eef9)
        /usr/local/go/src/net/http/client.go:187 +0x22
net/http.(*Client).do(0x0, 0xc0050ba400, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/http/client.go:527 +0x22c
net/http.(*Client).Do(...)
        /usr/local/go/src/net/http/client.go:509
cloud.google.com/go/storage.(*ObjectHandle).NewRangeReader.func2.1(0xc0056a12a8, 0xadd5c6)
        /go/src/github.com/cockroachdb/cockroach/vendor/cloud.google.com/go/storage/reader.go:137 +0x83
cloud.google.com/go/storage.runWithRetry.func1(0x0, 0x203003, 0x203001)
        /go/src/github.com/cockroachdb/cockroach/vendor/cloud.google.com/go/storage/invoke.go:28 +0x2a
cloud.google.com/go/internal.retry(0x62e01e0, 0xc00c403860, 0x0, 0x0, 0x0, 0x0, 0xc0056a1370, 0x57e50e0, 0xc0056a13a0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/vendor/cloud.google.com/go/internal/retry.go:38 +0x48
cloud.google.com/go/internal.Retry(...)
        /go/src/github.com/cockroachdb/cockroach/vendor/cloud.google.com/go/internal/retry.go:31
cloud.google.com/go/storage.runWithRetry(0x62e01e0, 0xc00c403860, 0xc0056a1498, 0x0, 0x203003)
        /go/src/github.com/cockroachdb/cockroach/vendor/cloud.google.com/go/storage/invoke.go:27 +0x87
cloud.google.com/go/storage.(*ObjectHandle).NewRangeReader.func2(0x0, 0xc00f5048c0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/vendor/cloud.google.com/go/storage/reader.go:136 +0x1b3
cloud.google.com/go/storage.(*ObjectHandle).NewRangeReader(0xc0015fec60, 0x62e0120, 0xc0011d0ac0, 0x0, 0xffffffffffffffff, 0x0, 0x203000, 0x0)
        /go/src/github.com/cockroachdb/cockroach/vendor/cloud.google.com/go/storage/reader.go:175 +0x637
github.com/cockroachdb/cockroach/pkg/storage/cloudimpl.(*resumingGoogleStorageReader).openStream.func1(0xc001bf65b0, 0xc0015fea50)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/cloudimpl/gcs_storage.go:190 +0x179
github.com/cockroachdb/cockroach/pkg/storage/cloudimpl.delayedRetry.func1(0xc0056a19f8, 0xc0011d0a01)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/cloudimpl/external_storage.go:415 +0x4d
github.com/cockroachdb/cockroach/pkg/util/retry.WithMaxAttempts(0x62e0120, 0xc0011d0ac0, 0x2faf080, 0x3b9aca00, 0x4000000000000000, 0x2, 0x0, 0x0, 0x3, 0xc0056a1ab0, ...)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:172 +0x10a
github.com/cockroachdb/cockroach/pkg/storage/cloudimpl.delayedRetry(0x62e0120, 0xc0011d0ac0, 0xc0056a1b38, 0x40, 0x4eff180)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/cloudimpl/external_storage.go:414 +0xfc
github.com/cockroachdb/cockroach/pkg/storage/cloudimpl.(*resumingGoogleStorageReader).openStream(0xc009a30700, 0xc009a30700, 0x2)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/cloudimpl/gcs_storage.go:188 +0x5a
github.com/cockroachdb/cockroach/pkg/storage/cloudimpl.(*gcsStorage).ReadFile(0xc009a306c0, 0x62e0120, 0xc0011d0ac0, 0x51ddec4, 0x6, 0x4, 0x6363e80, 0xc009a306c0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/cloudimpl/gcs_storage.go:242 +0x13b
github.com/cockroachdb/cockroach/pkg/ccl/backupccl.checkForExistingBackupsInCollection(0x62e0120, 0xc0011d0ac0, 0x6475e80, 0xc001dbe418, 0xc004af20b0, 0x1, 0x1, 0x1, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/backupccl/create_scheduled_backup.go:410 +0x19a
github.com/cockroachdb/cockroach/pkg/ccl/backupccl.doCreateBackupSchedules(0x62e0120, 0xc0011d0ac0, 0x6475e80, 0xc001dbe418, 0xc00119c500, 0xc0082d4480, 0xc001365668, 0x169eda1)
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/backupccl/create_scheduled_backup.go:300 +0xdf9
github.com/cockroachdb/cockroach/pkg/ccl/backupccl.createBackupScheduleHook.func1(0x62e0120, 0xc0011d0ac0, 0x0, 0x0, 0x0, 0xc0082d4480, 0xc001365740, 0xc0004641b0)
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/backupccl/create_scheduled_backup.go:670 +0x61
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1(0xc005f6c380, 0x62e0120, 0xc0011d0ac0, 0xc00a0b2840, 0xc001dbe418)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/planhook.go:146 +0x7d
created by github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/planhook.go:145 +0xcc
```

**Expected behavior**
I expected either a backup schedule to be created or an error to be returned.

**Environment:**
 - CockroachDB version: 20.2.2
 - Server OS: Ubuntu
 - Client app: cockroach sql

**Comment by dt** [comment]
Howdy stranger! Thanks for filing this!

I think this may have already been fixed by @miretskiy last week in https://github.com/cockroachdb/cockroach/pull/57651 so hopefully should be resolved in 20.2.3 which went out yesterday. Do you still see the issue with that version?

**Comment by a-robinson** [comment]
Hi! 👋

Sorry, I hadn't seen v20.2.3 and my search for previous issues apparently wasn't as exhaustive as it should have been. That does appear to fix it for me, thanks!
